tests/director/tests/bundle_instance_auth_test.go

```diff
@@ -159,10 +164,99 @@ func TestRequestBundleInstanceAuthCreationAsRuntimeConsumer(t *testing.T) {
 
 		// THEN
 		require.Error(t, err)
+		require.NotNil(t, output.RuntimeID)
+		require.Equal(t, runtime.ID, *output.RuntimeID)
+		require.Nil(t, output.RuntimeContextID)
 		require.Contains(t, err.Error(), "The operation is not allowed")
 	})
 }
 
+func TestRuntimeIdInBundleInstanceAuthIsSetToNullWhenDeletingRuntime(t *testing.T) {
+	ctx := context.Background()
+	tenantId := tenant.TestTenants.GetDefaultTenantID()
+
+	input := fixtures.FixRuntimeInput("runtime-test")
+	runtime := fixtures.RegisterRuntimeFromInputWithinTenant(t, ctx, dexGraphQLClient, tenantId, &input)
```

Comment:
I assume that you didn't put defer deleteRuntime here because you'll be doing that manually below, but still, if something breaks inbetween the runtime will be left as junk in DB and will probably cause conflict for the next test execution, because a runtime with name `runtime-test` already exists. So there has to be a defer in some form or manner, which could gracefully handle a 404 response, which will mean that the test actually finished entirely and already deleted the runtime.